chore: registry settings layout cleanup

deboer-tim opened the pull request:
### What does this PR do?

The registries page was three columns (location, user, then password & all actions) which caused odd scaling, esp. with the password column. The three types of registries (configured, suggested, and user added) also had different padding and margins for columns, which meant that editing a password would sometimes have touching controls or columns too wide to fit.

This just cleans it up into four consistent columns (location, user, password, and actions), with fixed widths (location 2x, all others 1x), and removes mix old or unused divs or flex classes.

### Screenshot / video of UI

Before:

https://github.com/containers/podman-desktop/assets/19958075/7c7c5a2a-5e6a-49d5-8253-1f62092af730

After:

https://github.com/containers/podman-desktop/assets/19958075/b1726cd4-49af-473a-8192-b1135558edde

### What issues does this PR fix or reference?

N/A

### How to test this PR?

Edit/add registries.

deboer-tim commented:
I'm pretty sure the original alignment was just by accident, and when you click on Configure the Login/Cancel don't line up at all. TBH I noticed too and wasn't sure what the fix should be, but I think it should be full alignment - maybe all the buttons right-justified?

benoitf commented:
I found https://github.com/containers/podman-desktop/issues/3233 and https://github.com/containers/podman-desktop/issues/4294 (but here problem is skipped ;-)

deboer-tim commented:
I had almost forgotten about #3233. Based on either link it looks like this should eventually be a Table, but based on the current implementation that could be a pain and not sure we should spend too much time atm.

jeffmaury commented:
I found strange the default layout when I switch to edit mode:

![image](https://github.com/containers/podman-desktop/assets/695993/c5f15e23-baee-4b8e-ae01-0c8b293073d7)

Seems to be that there is space for the Cancel button

deboer-tim commented:
The Cancel button is slightly wider than Login plus these is a small gap between them now. If you expand and contract the window a bit you can find the shift point, but I'm pretty sure the browser is correct :) and keep in mind the old behaviour was no gap and to let the Cancel button disappear under the right side of the window.

I played with the widths to give that column enough that it won't wrap until the window is fairly small, and to me it was handling this better than before - but at some point we're only going to see improvement if we rewrite the page with something like Table.

lstocchi commented:
I read the comments above. What about adding some flex justify center/end to the single actions cells? 
Something like 
https://drive.google.com/file/d/1FgTbskhnqxL3ucIB9BoXQSdIFT3fiIpR/view?usp=sharing

this is the edited code - there could be some hidden problems, just wanted to try quickly https://gist.github.com/lstocchi/06133861271f6ecbf97d11c83c39eceb

deboer-tim commented:
Rebased on main and added the flex as @lstocchi suggested, but went with left-justification to match the Configure/Login-Cancel left justification in other rows.

<img width="808" alt="Screenshot 2024-02-25 at 2 05 26 PM" src="https://github.com/containers/podman-desktop/assets/19958075/2ea411cd-4f0c-40fe-acff-7cfc7de25842">

benoitf reviewed (approved):
approving as it's better than what we have in main but I'm still not sure about that left alignment

jeffmaury reviewed (approved):
LGTM

deboer-tim commented:
> approving as it's better than what we have in main but I'm still not sure about that left alignment

I'll take a look at changing the alignment in another PR.

benoitf commented:
@deboer-tim you may wait for https://github.com/containers/podman-desktop/issues/4294 feedback
